File: CheckIn.Api.Common/Statistics/TaskSummaryStats.cs
using CheckIn.Api.Common.Enums;

namespace CheckIn.Api.Common.Statistics;

public class TaskSummaryStats
{
    public Guid TaskId { get; set; }
    public SubtaskMode Mode { get; set; }
    public double GlobalProgress { get; set; }

    // --- Nové polia pre detailnú vizualizáciu (Farby progress baru) ---

    /// <summary>
    /// Počet entít (osôb v Individual / subtaskov v Shared), ktoré sú kompletne dokončené VČAS.
    /// Táto hodnota napĺňa ZELENÚ časť progress baru.
    /// </summary>
    public int CompletedOnTime { get; set; }

    /// <summary>
    /// Počet entít (osôb v Individual / subtaskov v Shared), ktoré majú problém (neskoro splnené alebo aktuálne meškajú).
    /// Táto hodnota napĺňa ČERVENÚ časť progress baru.
    /// </summary>
    public int IssuesCount { get; set; }

    // --- Pôvodné polia (zostávajú pre spätnú kompatibilitu a textové labely) ---

    // Individual Mode
    public int TotalRespondents { get; set; }
    public int CompletedFull { get; set; } // Celkový počet ľudí, čo majú hotovo (bez ohľadu na včas/neskoro)
    public int InProgress { get; set; } // Začaté a nič nemešká (Oranžová časť)
    public int NotStarted { get; set; } // Nezačaté a nič nemešká (Sivá časť)

    // Shared Mode
    public int TotalSubtasks { get; set; }
    public int CompletedSubtasks { get; set; } // Čisto číselný údaj pre label X/Y
}

File: checkin_frontend/lib/core/models/Statistics/task_summary_stats.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import '../enums/task_enums.dart';

part 'task_summary_stats.freezed.dart';
part 'task_summary_stats.g.dart';

@freezed
sealed class TaskSummaryStats with _$TaskSummaryStats {
  const factory TaskSummaryStats({
    required String taskId,
    required SubtaskMode mode,
    required double globalProgress,

    @Default(0) int totalRespondents,
    @Default(0) int completedFull,
    @Default(0) int inProgress,
    @Default(0) int notStarted,
    @Default(0) int totalSubtasks,
    @Default(0) int completedSubtasks,
    @Default(0) int completedOnTime,
    @Default(0) int issuesCount,
  }) = _TaskSummaryStats;

  factory TaskSummaryStats.fromJson(Map<String, dynamic> json) =>
      _$TaskSummaryStatsFromJson(json);
}